### PR TITLE
#1280: Add some functional tests to check that modules can be uninstalled and reinstalled without errors.

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Adding some tests to make sure our new az_select_menu and az_global_footer modules are uninstallable and reinstallable.

A missing functional regression test for https://github.com/az-digital/az_quickstart/pull/1281 as part of https://github.com/az-digital/az_quickstart/issues/1280

Related to #1318

### CODE REVIEW

**Comment by trackleft** [comment]
Tested this by undoing https://github.com/az-digital/az_quickstart/pull/1281

image

File: modules/custom/az_global_footer/tests/src/Functional/AzGlobalFooterTest.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    parent::setUp();
+
+    // Create a test user.
+    $this->adminUser = $this->drupalCreateUser([
+      'access administration pages',
+      'administer modules',
+    ]);
+    $this->drupalLogin($this->adminUser);
+  }
+
```
Comment (trackleft) [inline modules/custom/az_global_footer/tests/src/Functional/AzGlobalFooterTest.php:61]:
```suggestion

```

File: modules/custom/az_select_menu/tests/src/Functional/AzSelectMenuTest.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  protected function setUp() {
+    parent::setUp();
+
+    // Create a test user.
+    $this->adminUser = $this->drupalCreateUser([
+      'access administration pages',
+      'administer modules',
+    ]);
+    $this->drupalLogin($this->adminUser);
+  }
```
Comment (trackleft) [inline modules/custom/az_select_menu/tests/src/Functional/AzSelectMenuTest.php:60]:
```suggestion
```

File: modules/custom/az_select_menu/tests/src/Functional/AzSelectMenuTest.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+   * @var bool
+   */
+  protected $strictConfigSchema = FALSE;
+
+  /**
+   * Tests that the Quickstart Select Menu Block module can be uninstalled.
+   *
+   * @group az_select_menu
+   */
+  public function testIsUninstallable() {
```
Comment (trackleft) [inline modules/custom/az_select_menu/tests/src/Functional/AzSelectMenuTest.php:38]:
```suggestion
  public function testIsUninstallableAndReinstallable() {
```

**Review by joeparsons — APPROVED** [review]
This looks good to me now.

**Review by bberndt-uaz — APPROVED** [review]

**Pull request merged**